drivers/stmpe811/stmpe811_touch_dev.c

```diff
@@ -57,8 +60,13 @@ uint8_t _stmpe811_touches(const touch_dev_t *touch_dev, touch_t *touches, size_t
     if (ret && touches != NULL) {
         stmpe811_touch_position_t pos;
         stmpe811_read_touch_position(dev, &pos);
-        touches[0].x = pos.x;
-        touches[0].y = pos.y;
+        /* STMPE811 driver returns the position with origin at the bottom left
+        corner and portrait orientation, so convert them to use top left corner
+        as origin and landscape orientation. */
+        touches[0].x = pos.y;
+        touches[0].y = dev->params.xmax - pos.x;
```

Comment:
I feel this should be configurable and screens might be wired up differently, but we can leave that for a later PR.